Ethereal Artefacts/Ethereal Artefacts/Views/LoginView.swift

```diff
@@ -0,0 +1,129 @@
+                            if !isEditing {
+                                isEmailValid = validateEmail(userEmail.lowercased())
+                            }
+                        })
+                        .padding()
+                        .background(Color.white)
+                        .foregroundColor(isEmailValid ? Color(.black) : Color.red)
+                        .cornerRadius(10)
+                        .padding(.bottom, 10)
+                        
+                        HStack {
+                            if showPassword {
+                                TextField("Password", text: $password)
+                            } else {
+                                SecureField("Password", text: $password)
+                            }
+                            Button(action: { showPassword.toggle() }) {
+                                Image(systemName: showPassword ? "eye.fill" : "eye.slash.fill")
+                                    .foregroundColor(.gray)
+                            }
+                        }
+                        .padding()
+                        .background(Color.white)
+                        .cornerRadius(10)
+                        .padding(.bottom, 10)
+                        
+                        Button(action: {
+                            if userEmail == validUserEmail && password == validPassword {
+                                loginError = nil
+                                isLoggedIn = true
+                            } else {
+                                loginError = "Invalid email address."
+                                password = ""
+                            }
+                        }) {
+                            Text("Log in")
+                                .font(.headline)
+                                .fontWeight(.bold)
+                                .foregroundColor(.white)
+                                .padding()
+                                .frame(minWidth: 20, maxWidth: .infinity)
+                                .background(Color(hex: "47337A"))
+                                .cornerRadius(24)
+                        }
+                        .padding()
+                        
+                        if let error = loginError {
+                            Text(error)
+                                .foregroundColor(.red)
+                        }
+                        
+                        Spacer()
+                    }
+                    .padding()
+                }
+                .padding(.top, 110)
+            }
+            .background(
+                NavigationLink(
+                    destination: HomepageView().navigationBarBackButtonHidden(true),
+                    isActive: $isLoggedIn
+                ) {
+                    EmptyView()
+                }
+            )
+        }
+    }
+}
+
+extension Color {
```

Comment:
Best approach here is to create color asset in Assets and then use like this for example:  Color("MidnightBlue")